logmein_host/logmein_host.py

```diff
@@ -410,6 +476,21 @@ def __handleDataSock(self, data):
         buf.readline() # clientAddr
         connectionId = buf.readline().rstrip()
 
+        # assign connection to session
+        if hostAuth in self.sessions.keys():
+            session = self.sessions[hostAuth]
+            session.onActivity()
+            self.__assignConnIdToSession(connectionId, session.id)
+        else:
+            self.pendingRequests.append(Session(hostAuth))
```

Comment:
Here you put a `Session` object to `pendingRequests` but in `__handleSessionData` you read as a host auth code.